release-gap-package

```diff
@@ -605,7 +605,7 @@ git commit -m "Update website for $PKG $VERSION"
 
 if [ "x$PUSH" = xyes ] ; then
     notice "Pushing website changes"
-    git push
+    git push "$REMOTE"
```

Comment:
Hmm, I haven't test it now, but IIRC this only will work for certain values of `push.default` in the git config -- certainly for the old pre-2.0 default mode `matching`. But with the post-2.x default of `current`, won't it still only work if an explicit upstream has been configured?

Of course one can avoid this by specifying an explicit destination branch -- but then we have to hardcode `gh-pages` here, which also comes with its share of issues. Hm
```suggestion
    git push "$REMOTE" gh-pages
```

Reply:
I have tested this both locally and in a CI environment and have thus deemed it "reasonably safe", but I see the possible problems.

I could get the branch name and use that to push as in
https://github.com/gap-system/ReleaseTools/blob/b5182b2562f0102e17e95d53ebe88e1b97cf490e/release-gap-package#L477

Then at least the pushes are consistent :-)

Reply:
ok, here's what I'd suggest: let's indeed push the branch, but put it into an env var like `WEBBRANCH` that  defaults to `gh-pages` but can be overridden. Then there is at least an escape hatch for power users (and I am pretty sure it'll be only those, if any, that aren't using `gh-pages` anyway)

Reply:
We could also just do the same as we did to compute branch, just inside the web dir, i.e.
```suggestion
    WEBBRANCH=$(git symbolic-ref -q --short HEAD)
    git push "$REMOTE" "$WEBBRANCH"
```
The drawback of that is that it hardcodes another assumption (that the local and remove branch have the same name), and cannot be overridden. But one could use it instead of just `gh-pages` as default -- as long as there is a way to override it.

Reply:
> The drawback of that is that it hardcodes another assumption (that the local and remove branch have the same name)

The same statement holds true for pushing the main (i.e. non-web) branch in https://github.com/gap-system/ReleaseTools/blob/b5182b2562f0102e17e95d53ebe88e1b97cf490e/release-gap-package#L477 or am I missing something? If nobody needs an override there, I would not expect someone to need an override here. So maybe we could defer an override to the future in case anyone actually needs it, and in that case implement it for both pushes (i.e., the non-web branch and the web branch)?

Reply:
I analyzed this again and in more details, and I now think this PR is OK to merge as-is; if somebody complains, we'll revise it.

Sorry for taking so long to do this, but I simply forgot about this PR :-((((

Reply:
> Sorry for taking so long to do this, but I simply forgot about this PR :-((((

No problem! Just a note: I have more changes I could create PRs for (for example uploading PDFs), but this is blocked on the discussion in #70 (this is in no way urgent for me, just things I would like to upstream in the long run).